src/librustc/dep_graph/graph.rs

```diff
@@ -128,15 +171,18 @@ impl DepGraph {
     }
 
     pub fn query(&self) -> DepGraphQuery {
-        let current = &self.data.as_ref().unwrap().current;
+        let data = self.data.as_ref().unwrap();
+        let current = &data.current;
         let shards = current.data.lock_shards();
-        let node_count = current.node_count.load(Acquire) as usize;
+        let node_count = data.node_count();
         let data: IndexVec<DepNodeIndex, _> = (0..node_count).map(|i| {
             let shard = i % sharded::SHARDS;
             let inner = i / sharded::SHARDS;
             &shards[shard][inner]
         }).collect();
 
+        // FIXME: Only use Red/Green nodes here.
```

Comment:
what does this FIXME mean?

Reply:
It's the reason for the `[WIP]`. The graph created here can have nodes from the previous session while this is supposed to create the dep graph from the current session for testing purposes. So this is missing some code to strip the unknown  and invalid nodes.